client/src/test/scala/dwolla/cloudflare/AccessControlRuleClientSpec.scala

```diff
@@ -0,0 +1,338 @@
+            mode = Some("challenge"),
+            configuration = RuleConfiguration("ip", "1.2.3.4"),
+            createdOn = Some("2014-01-01T05:20:00.12345Z"),
+            modifiedOn = Some("2014-01-01T05:20:00.12345Z"),
+            scope = RuleScope("fake-rule-scope", "Some Account", "account")
+          )
+        )
+      )
+    }
+  }
+
+  "create" should {
+    "create new access rule" in new Setup {
+      val rateLimitId = "fake-access-rule-1"
+
+      val createRateLimit = CreateRule(Option("challenge"), RuleConfiguration("ip", "1.2.3.4"), Option("Some notes"))
+
+      val http4sClient = fakeService.client(fakeService.createAccessRule(SampleResponses.Successes.accessRule, accountId))
+      val client = buildAccessControlRuleClient(http4sClient, authorization)
+
+      val output = client.create(accountId, createRateLimit)
+        .compile.toList.unsafeRunSync()
+
+      output must be_==(
+        List(
+          Rule(
+            id = "fake-access-rule-1",
+            notes = Some("Some notes"),
+            allowedModes = List("whitelist", "block", "challenge", "js_challenge"),
+            mode = Some("challenge"),
+            configuration = RuleConfiguration("ip", "198.51.100.4"),
+            createdOn = Some("2014-01-01T05:20:00.12345Z"),
+            modifiedOn = Some("2014-01-01T05:20:00.12345Z"),
+            scope = RuleScope("fake-rule-scope", "Some Account", "account")
+          )
+        )
+      )
+    }
+
+    "throw unexpected exception if error creating access rule" in new Setup {
+      val createAccessRule = CreateRule(Option(""), RuleConfiguration("ip", "198.51.100.4"), Option("Some notes"))
+
+      val failure = SampleResponses.Failures.accessRuleCreationErrorMode
+      val http4sClient = fakeService.client(fakeService.createAccessRule(failure.json, accountId, failure.status))
+      val client = buildAccessControlRuleClient(http4sClient, authorization)
+
+      val output = client.create(accountId, createAccessRule)
+        .compile
+        .toList
+        .attempt
+        .unsafeRunSync()
+
+      output must beLeft[Throwable].like {
+        case ex: UnexpectedCloudflareErrorException ⇒ ex.getMessage must_==
+          """An unexpected Cloudflare error occurred. Errors:
+            |
+            | - Error(None,firewallaccessrules.api.validation_error:unsupported target received: ,invalid mode received: )
+            |     """.stripMargin
+      }
+    }
+  }
+
+  "deleteAccessRule" should {
+    "delete access rule from account" in new Setup {
+      val ruleId = "fake-access-rule-1"
+
+      val http4sClient = fakeService.client(fakeService.deleteAccessRule(SampleResponses.Successes.removedAccessRule, accountId, ruleId))
+      val client = buildAccessControlRuleClient(http4sClient, authorization)
+
+      val output = client.delete(accountId, ruleId)
+        .compile.toList.unsafeRunSync()
+
+      output must be_==(List(ruleId))
```

Comment:
This could be `output must contain(ruleId)` instead of `output must be_==(List(ruleId))`